plugins/system/shortcut/fields/keyselect.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @package     Joomla.Plugins
+ * @subpackage  System.shortcut
+ *
+ * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die('Restricted access');
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\HTML\HTMLHelper;
+
+jimport('joomla.form.formfield');
+
+class JFormFieldKeyselect extends JFormField  {
+	
+	protected $type = 'Keyselect';
+
+	public function getInput() {
+		$return = '';
+		
+		HTMLHelper::_('bootstrap.modal');
+		
+		$app = Factory::getApplication();
+		Factory::getDocument()->addScriptOptions('modal_header_title', 'Set Shortcut');
+		Factory::getDocument()->addScriptOptions('modal_combination_text', 'Current Combination:');
+
+		$wa = $app->getDocument()->getWebAssetManager();
```

Comment:
its showing 
![image](https://user-images.githubusercontent.com/30978328/124442578-81f5be00-dd9a-11eb-80e1-78083d36d132.png)

undefined method @HLeithner Please guide here

Reply:
Your idea tells you that the object `CMSApplicationInterface` doesn't have a function `getDocument()` which is actually true. But you get an object extended from `WebApplication` which actually has this method.

Reply:
$wa needs one more space before the =